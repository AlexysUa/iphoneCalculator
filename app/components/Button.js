import React from 'react';
import { StyleSheet, TouchableOpacity, Text } from 'react-native';

function Button({ onPressHandler, value }) {
  const checkOperationButton = (buttonValue) => {
    if(['+','=','-','x','m+','รท'].includes(buttonValue)) {
      return styles.operationButton;
    }
    if(buttonValue === '0') {
      return {flex: 2, alignItems: 'flex-start', paddingLeft: 28}
    }
    if(['AC', `+/-`, '%'].includes(buttonValue)){
      return styles.specialButton;
    }    
  }  

  return (
    <TouchableOpacity
      style={[styles.button, checkOperationButton(value)]}
      onPress={onPressHandler}      
    > 
      <Text style={styles.buttonText}>{value}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    flex: 1,
    borderRadius: 100,    
    backgroundColor: 'hsl(0, 0%, 17%)',    
    justifyContent: 'center',
    alignItems: 'center', 
    marginTop: 4,
    marginRight: 8,
    marginBottom: 4,
    marginLeft: 8,       
  },
  buttonText: {
    color:'white',
    fontSize: 32,
  },
  operationButton: {
    backgroundColor: '#FA9307',
  },
  specialButton: {
    backgroundColor: '#A4A4A4',
  }
})

export default Button;