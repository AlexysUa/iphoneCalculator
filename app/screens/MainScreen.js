import React, { useState } from 'react';
import { Text, StyleSheet, View, SafeAreaView } from 'react-native';

import Button from '../components/Button';

const buttons = [
  ['AC', `+/-`, '%', 'รท'],
  ['mc', 'mr', 'm-', 'm+'],
  ['7', '8', '9', 'x'],
  ['4', '5', '6', '-'],
  ['1', '2', '3', '+'],
  ['0', '.', '=']
]

function MainScreen() {
  const [outputValue, setOutputValue] = useState('0');    
  const [pointChecker, setPointChecker] = useState(false);
  const [calcResult, setCalcResult] = useState(''); 
  const [calcMemory, setCalcMemory] = useState([]);  

  const addAllButtons = () => {
    let allButtons = buttons.map((buttonsRows, index) => {
      let rows = buttonsRows.map((eachButton, buttonIndex) => {        
        return (
          <Button 
            key={'button' + buttonIndex} 
            value={eachButton}
            onPressHandler={handlerForButton.bind(this, eachButton)}
          />
        )
      });

      return <View key={'row' + index} style={styles.buttonRow}>{rows}</View>
    });

    return allButtons;
  }
  const resultChecker = (result, operator = '') => {
    let displayResult = eval(result).toString();    
    
    if(!/\./g.test(displayResult) && displayResult.length > 9) {
      let expoResult = eval(result).toExponential(1).toString().replace(/.0|\+/g, '');
      setOutputValue(expoResult);
      setCalcResult(expoResult + operator);
    } else {
      let numbersPart = displayResult.split(/\./);
      let formattedResult = +(eval(displayResult).toFixed(numbersPart[0].length>10 ? 0 : 10 - numbersPart[0].length));        
        setOutputValue(formattedResult);
        setCalcResult(formattedResult + operator); 
    }    
  }   

  const handlerForButton = (buttonInput) => {     
    let res; 
    switch (buttonInput) {      
      case '0':
      case '1':
      case '2':
      case '3':
      case '4':
      case '5':
      case '6':
      case '7':
      case '8':
      case '9':                
        if(outputValue.length >= 9 && /^\d+$/.test(calcResult) || /[\+\-\*\/]0$/.test(calcResult)) {                  
          break;
        }
        if(/\d+\.?\d+$/.test(calcResult) && outputValue.length >= 9) {
          break;
        }
        if(/[\+\-\*\/]/.test(calcResult[calcResult.length-1])) {          
          setOutputValue('');                    
        }        
        setOutputValue(current=>(current === '0' || current === '-0') ? buttonInput : current + buttonInput);               
        setCalcResult(current=>(current === '0') ? buttonInput : current + buttonInput);         
        break;               
      case 'AC':
        setOutputValue('0');
        setCalcResult('');
        setPointChecker(false);
        break;
      case 'รท':        
      case 'x':         
      case '-':        
      case '+':
        setCalcResult(calcResult)
        setPointChecker(false);
        let formattedOperator = (buttonInput === 'x') ? '*' : (buttonInput === 'รท') ? '/' : buttonInput;          
        if(/\w+[\+\-\*\/]\w+/.test(calcResult)) {
          res = eval(calcResult.replace(/0*$/,"")).toString().substr(0,9);
          resultChecker(calcResult, formattedOperator)            
          break;
        } 
        if(!/[\+\-\*\/]/.test(calcResult[calcResult.length-1])) {
          setCalcResult(calcResult+formattedOperator);            
        }
        if(/\d+[\+\-\*\/]$/.test(calcResult)) {
          setCalcResult(current=>current.substr(0, current.length - 1) + formattedOperator);            
        }        
        break;
      case '.':
        if(outputValue.includes('.') && !pointChecker) {            
          setPointChecker(true);
          break;
        }                       
        setOutputValue(pointChecker ? outputValue : outputValue + buttonInput);
        setCalcResult(pointChecker ? calcResult : calcResult + buttonInput)
        break;
      case '=':
        if(/^\d+[\+\-\*\/]$/.test(calcResult) || calcResult === '0') {
          break;
        }
        if(!/[\+\-\*\/]/.test(calcResult[calcResult.length-1])) {
          setCalcResult(calcResult+formattedOperator);            
        }                    
        if(/^\d+\.?\d?[\+\-\*\/]\d+$/.test(calcResult) || /\d+/.test(calcResult)) {                                    
          res = eval(calcResult.replace(/\.0*$/,"")).toString();       
        }
        setOutputValue(res);  
        setCalcResult(res); 
        setPointChecker(false);                
        resultChecker(calcResult);
        break;
      case `+/-`:
        let insertedMinus = calcResult;
        
        if(/^\-?\d*\.?\d+$/.test(calcResult) || calcResult.includes('e')) {          
          if(calcResult.includes('-')) {            
            setCalcResult(current=> current.substr(1,current.length-1));
            setOutputValue(current=> current.substr(1,current.length-1));
          } else {
            setOutputValue(current=>'-' + current)
            setCalcResult(current=> '-' + current);
          }
        }
        if(/^\-?\d+\.?\d+[\+\-\*\/]\(?\-?\d+\.?\d+\)?$/.test(calcResult)) {          
          if(/\({1}\-{1}\d+\.?\d?\){1}$/.test(calcResult)) {
            console.log('tut')
            let cutMinus = calcResult.replace(/\(\-?|\)/g,'');
            setCalcResult(cutMinus);
            setOutputValue(current=>current.replace(/\-/g,''));            
          } else {            
            insertedMinus = insertedMinus.substr(0,calcResult.length-outputValue.length) + '('  + '-' + insertedMinus.substr(calcResult.length-outputValue.length) + ')';            
            setCalcResult(insertedMinus);
            setOutputValue(current=>'-' + current);
          }         
          
        }        
        break;  
      case '%':
        if(/^\-?\d*\.?\d+$/.test(calcResult)){
          setCalcResult(current=> (current/100).toString().replace(/\.0*$/,""));
          setOutputValue(current=> (current/100).toString().replace(/\.0*$/,""));
          break;
        }
        if(/^\-?\d+\.?\d+[\+\-\*\/]\(?\-?\d+\.?\d+\)?$/.test(calcResult)) {
          let indexSecondNumber = calcResult.lastIndexOf(outputValue);
          let newNumber = ((+calcResult.slice(indexSecondNumber, indexSecondNumber+ outputValue.length))/100).toString();
          let resultNumber = calcResult.substr(0,indexSecondNumber) + `(${newNumber})`;
          let currentValue = (outputValue/100).toString().replace(/\.0*$/,"");
          if(/^\-?\d+\.?\d+[\+\-\*\/]\({1}\-?\d+\.?\d+\){1}$/.test(calcResult)) {
            resultNumber = calcResult.substr(0,indexSecondNumber-1) + `(${newNumber})`;
          }            
          setCalcResult(resultNumber);
          setOutputValue(currentValue); 
        }
      case 'mr':
        let resultMemory = (eval(calcMemory.join('+'))).toString().replace(/\.0*$/,"");
        setCalcResult(resultMemory);
        setOutputValue(resultMemory);
        break;
      case 'mc': 
        setCalcMemory([]);
        break;
      case 'm-': 
        if(/\d+\.?\d*[\+\-\*\/]?\(?\d+\.?\d*\)?$/.test(calcResult) || /^-?\d+$/.test(calcResult)) {
          calcMemory.pop();          
          if(calcMemory.length>0) {
            break;
          }
          setCalcMemory(calcMemory);
        } 
        break;
      case 'm+': 
        if(/\d+\.?\d*[\+\-\*\/]?\(?\d+\.?\d*\)?$/.test(calcResult) || /^-?\d+$/.test(calcResult)) {
          setCalcMemory(current=> [...current, `(${calcResult})`])
        }                 
        break;  
      }            
    }   
    
    return (    
      <SafeAreaView style={styles.container}>
        <View style={styles.inputContainer}>          
          <Text style={styles.inputText}>{outputValue}</Text>
        </View>
        <View style={styles.buttonsContainer}>
          {addAllButtons()}
          {console.log(calcMemory, calcResult)}
        </View>      
      </SafeAreaView>
    );
  }
  
  const styles = StyleSheet.create({
    container: {
      flex: 1,    
      backgroundColor: 'black',    
    },  
    inputContainer: {
      justifyContent:'flex-end',
      alignItems: 'flex-end',
      flex: 2,
      margin: 20,
    },
    inputText: {
      fontSize: 60,
      color:'white',
      textAlign: 'right',
    },
    buttonsContainer: {
      flex: 8,
    },
    buttonRow: {
      flex: 1,    
      flexDirection: 'row',
      justifyContent: 'space-around',    
    }  
  })
  
  export default MainScreen; 