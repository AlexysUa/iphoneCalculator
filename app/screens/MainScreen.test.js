import React from 'react';
import renderer from 'react-test-renderer';

import MainScreen from './MainScreen';
import Button from '../components/Button';

describe('<MainScreen />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<MainScreen />).toJSON();
    expect(tree.children.length).toBe(2);
  });
});

it("renders Button", () => {
  const tree = renderer.create(<Button />).toJSON();

  expect(tree).toMatchSnapshot();
});
