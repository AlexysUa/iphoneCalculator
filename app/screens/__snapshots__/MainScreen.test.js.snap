// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "hsl(0, 0%, 17%)",
      "borderRadius": 100,
      "flex": 1,
      "justifyContent": "center",
      "marginBottom": 4,
      "marginLeft": 8,
      "marginRight": 8,
      "marginTop": 4,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 32,
      }
    }
  />
</View>
`;

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 2,
        "justifyContent": "flex-end",
        "margin": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 60,
          "textAlign": "right",
        }
      }
    >
      0
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 8,
        "margin": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#A4A4A4",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          AC
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#A4A4A4",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          +/-
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#A4A4A4",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          %
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FA9307",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          รท
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "hsl(0, 0%, 17%)",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          mc
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "hsl(0, 0%, 17%)",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          mr
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "hsl(0, 0%, 17%)",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          m-
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FA9307",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          m+
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "hsl(0, 0%, 17%)",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          7
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "hsl(0, 0%, 17%)",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          8
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "hsl(0, 0%, 17%)",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          9
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FA9307",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          x
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "hsl(0, 0%, 17%)",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          4
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "hsl(0, 0%, 17%)",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          5
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "hsl(0, 0%, 17%)",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          6
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FA9307",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          -
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "hsl(0, 0%, 17%)",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          1
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "hsl(0, 0%, 17%)",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          2
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "hsl(0, 0%, 17%)",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          3
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FA9307",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          +
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "flex-start",
            "backgroundColor": "hsl(0, 0%, 17%)",
            "borderRadius": 100,
            "flex": 2,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
            "paddingLeft": 28,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          0
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "hsl(0, 0%, 17%)",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          .
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FA9307",
            "borderRadius": 100,
            "flex": 1,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 4,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
            }
          }
        >
          =
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`renders correctly 2`] = `
<Text>
  Snapshot test!
</Text>
`;
